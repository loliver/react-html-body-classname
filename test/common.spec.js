import React from 'react'
import { render } from '@testing-library/react'
import '@testing-library/jest-dom/extend-expect'

import { HtmlClassName, BodyClassName } from '../'

describe('HtmlClassName', () => {
  global.beforeEach(() => {
    HtmlClassName.canUseDOM = false
  })

  it('hides itself from the DOM', () => {
    var { container } = render(
      <HtmlClassName className='irrelevant'>
        <div>hello</div>
      </HtmlClassName>
    )
    expect(container).toMatchSnapshot()
  })

  it('throws an error if it has multiple children', () => {
    var Component = () => (
      <HtmlClassName className='irrelevant'>
        <div>hello</div>
        <div>world</div>
      </HtmlClassName>
    )

    expect(() => {
      render(<Component />)
    }).toThrow(
      /React.Children.only expected to receive a single React element child/
    )
  })

  it('works with complex children', () => {
    var Component1 = () => {
      return (
        <p>
          <span>c</span>
          <span>d</span>
        </p>
      )
    }

    var Component2 = () => {
      return (
        <HtmlClassName className='irrelevant'>
          <div>
            <div>a</div>
            <div>b</div>
            <div>
              <Component1 />
            </div>
          </div>
        </HtmlClassName>
      )
    }

    var { container, getByText } = render(<Component2 />)
    expect(container).toMatchSnapshot()
    expect(getByText('a')).toBeDefined()
    expect(getByText('b')).toBeDefined()
    expect(getByText('c')).toBeDefined()
    expect(getByText('d')).toBeDefined()
  })
})

describe('BodyClassName', () => {
  global.beforeEach(() => {
    BodyClassName.canUseDOM = false
  })

  it('hides itself from the DOM', () => {
    var { container } = render(
      <BodyClassName className='irrelevant'>
        <div>hello</div>
      </BodyClassName>
    )
    expect(container).toMatchSnapshot()
  })

  it('throws an error if it has multiple children', () => {
    var Component = () => (
      <BodyClassName className='irrelevant'>
        <div>hello</div>
        <div>world</div>
      </BodyClassName>
    )

    expect(() => {
      render(<Component />)
    }).toThrow(
      /React.Children.only expected to receive a single React element child/
    )
  })

  it('works with complex children', () => {
    var Component1 = () => (
      <p>
        <span>c</span>
        <span>d</span>
      </p>
    )

    var Component2 = () => (
      <BodyClassName className='irrelevant'>
        <div>
          <div>a</div>
          <div>b</div>
          <div>
            <Component1 />
          </div>
        </div>
      </BodyClassName>
    )

    var { container, getByText } = render(<Component2 />)
    expect(container).toMatchSnapshot()
    expect(getByText('a')).toBeDefined()
    expect(getByText('b')).toBeDefined()
    expect(getByText('c')).toBeDefined()
    expect(getByText('d')).toBeDefined()
  })
})
