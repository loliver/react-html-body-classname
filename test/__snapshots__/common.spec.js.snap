// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BodyClassName hides itself from the DOM 1`] = `
<div>
  <div>
    hello
  </div>
</div>
`;

exports[`BodyClassName works with complex children 1`] = `
<div>
  <div>
    <div>
      a
    </div>
    <div>
      b
    </div>
    <div>
      <p>
        <span>
          c
        </span>
        <span>
          d
        </span>
      </p>
    </div>
  </div>
</div>
`;

exports[`HtmlClassName hides itself from the DOM 1`] = `
<div>
  <div>
    hello
  </div>
</div>
`;

exports[`HtmlClassName works with complex children 1`] = `
<div>
  <div>
    <div>
      a
    </div>
    <div>
      b
    </div>
    <div>
      <p>
        <span>
          c
        </span>
        <span>
          d
        </span>
      </p>
    </div>
  </div>
</div>
`;
